src/translate_c.zig

```diff
@@ -2357,20 +2357,176 @@ fn transCCast(
             .rhs = try Tag.bit_cast.create(c.arena, expr),
         });
     }
-    if (cIsInteger(dst_type) and qualTypeIsPtr(src_type)) {
-        // @intCast(dest_type, @intFromPtr(val))
-        const int_from_ptr = try Tag.int_from_ptr.create(c.arena, expr);
-        return Tag.as.create(c.arena, .{
-            .lhs = dst_node,
-            .rhs = try Tag.int_cast.create(c.arena, int_from_ptr),
+    if (qualTypeIsPtr(src_type) and cIsInteger(dst_type)) {
+        // Steps:
+        // 1. @intFromPtr(src_ptr)
+        // 2. Truncate if pointer width is wider than dst int
+        // 3. Cast to unsigned int of destination size
+        // 4. If dest int is signed, bitcast.
+        //
+        // We don't know if the dest int type has a shorter width than the pointer,
+        // until compile-time, so we need to add a check in the generated code
+        // in case a @truncate() cast is needed.
+        //
+        // Example:
+        //  (blk: {
+        //      if(@sizeOf(src_ptr_type) > @sizeOf(dst_int_type)) {
+        //          break :blk @as(c_ulong, @truncate(@intFromPtr(expr)));
+        //      }
+        //      else {
+        //          break :blk @as(c_ulong, @intFromPtr(expr));
+        //      }
+        //      break blk: _tmp
```

Comment:
This `break blk: _tmp` line seems like it was inadvertently included in the comment?

Reply:
Good catch, I'll remove that.